### DISCUSSION TITLE
Can't find "mip"

### DISCUSSION THREAD

**Discussion by ctimmer** [micropython/micropython#9567] [Libraries & Drivers] [answered]
Just read an interesting [discussion](https://github.com/orgs/micropython/discussions/9565) about "mip" and "upip". "mip" doesn't seem to be included with the latest firmware I am using (esp32-20220618-v1.19.1.bin). Is there something I need to do to enable/install it? Looks like a great utility.

Thanks for the help,
Curt

**Comment by jimmo** [comment, marked as answer]
It's in the nightly / "unstable" builds since last week, and will be in the upcoming v1.20 release firmware.

It's also in mpremote since the v0.4.0 release last week.

https://docs.micropython.org/en/latest/reference/packages.html

**Reply by ctimmer** [reply]
Thanks for the clarification

**Reply by bkyle** [reply]
I found it very confusing that the latest documentation doesn't indicate that it's for the unstable/nightly builds.  I assumed it was for the latest stable version as indicated by the version number in the documentation.  I don't know how frequently the are discrepancies between what's in the "latest" docs and what's available in the latest stable builds but I seem to recall getting bitten by this before.

**Reply by jimmo** [reply]
@bkyle I agree!  I raised https://github.com/micropython/micropython/pull/9668 to address this.

**Reply by scruss** [reply]
this happens all the time; I've had to advise two or three folks this week alone.

You want to refer to [v1.19.1](https://docs.micropython.org/en/v1.19.1/), not [latest](https://docs.micropython.org/en/latest/index.html).
